iml-agent/src/rpm.rs

```diff
@@ -1,25 +1,89 @@
 use regex::Regex;
 
+use std::fmt;
 use std::process::Output;
+use std::result::Result as StdResult;
 
 use crate::{agent_error::ImlAgentError, cmd::cmd_output};
 
-fn parse(output: Output) -> Result<bool, ImlAgentError> {
+enum PackageState {
+    Installed(Version),
+    NotInstalled,
+}
+
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Version(String);
+
+impl fmt::Display for Version {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum Result {
```

Comment:
It looks the same as `Result<Version, Error>`, maybe use it?

Reply:
Yeah, I didn't exactly want to define a type, but rather not deal with generic `Result<T, E>` all the way down.

Made it an alias instead.